CLA StreamsLookup ErrorHandler (#1795)
smartcontractkit/documentation: src/features/chainlink-automation/common/StreamsLookupInterface.mdx

| -------------- | --------- | ----------------------------------------------------------------------------------------------------------------------------------------------- | ------------------ |
| `performData`  | `bytes`   | Data encoded in the `oracleCallback` function that will be used to execute function onchain, e.g. encode signed reports and log data if need be | `bytes`            |
| `upkeepNeeded` | `boolean` | When set to `true` this will trigger the `performUpkeep` function to be executed onchain                                                        | `true` of `false`  |

### checkErrorHandler function

> This function is simulated offchain to respond to [error codes](/chainlink-automation/guides/streams-lookup-error-handler#error-codes) before sending data onchain via `performUpkeep`. 

Reviewer: ```suggestion
This function is simulated offchain to respond to [error codes](/chainlink-automation/guides/streams-lookup-error-handler#error-codes) before sending data onchain in the `performUpkeep` function. 
```

#### Parameters

| Variable Name | Type      | Description                                                    | Permissible Values                                                             |
| ------------- | --------- | -------------------------------------------------------------- | ------------------------------------------------------------------------------ |

> | `errorCode`      | `uint` | Error code returned by StreamsLookup error handler | e.g. "800400" (`ErrCodeStreamsBadRequest`) |

Reviewer: ```suggestion
| `errorCode`      | `uint` | Error code returned by StreamsLookup error handler | Example: "800400" (`ErrCodeStreamsBadRequest`) |
```

> | `extraData`   | `bytes`   | Bytes data sent as part of original MercuryLookup revert error | e.g. can send log data if you want to decode to use                            |

Reviewer: ```suggestion
| `extraData`   | `bytes`   | Bytes data sent as part of the original `MercuryLookup` revert error | This field can send log data if you want to decode it and use it.                            |
```

#### Outputs

| Variable Name  | Type      | Description                                                                                                                                     | Permissible Values |
| -------------- | --------- | ----------------------------------------------------------------------------------------------------------------------------------------------- | ------------------ |
| `performData`  | `bytes`   | Data encoded in the `oracleCallback` function that will be used to execute function onchain, e.g. encode signed reports and log data if need be | `bytes`            |

> | `upkeepNeeded` | `boolean` | When set to `true` this will trigger the `performUpkeep` function to be executed onchain                                                        | `true` of `false`  |

Reviewer: ```suggestion
| `upkeepNeeded` | `boolean` | When set to `true`, this triggers the `performUpkeep` function to be executed onchain                                                        | `true` or `false`  |
```
